client/web/src/user/settings/auth/ExternalAccountsSignIn.tsx

```diff
@@ -95,6 +116,33 @@ const getNormalizedAccount = (
                     }
                 }
                 break
+            case 'SAML':
+                {
+                    const samlExternalData = accountExternalData as SamlExternalData
+                    const data = samlExternalData.Assertions[0].AttributeStatement.Attributes
+                    let nickname = ''
+
+                    for (let index = 0, length = data.length; index < length; index += 1) {
+                        const item = data[index]
+                        if (data[index].Values && item.Name) {
+                            const key = item.Name.slice(Math.max(0, item.Name.lastIndexOf('/') + 1))
+                            if (key === 'nickname') {
+                                nickname = item.Values[0].Value
+                            }
+                        }
+                    }
+
+                    normalizedAccount = {
+                        ...normalizedAccount,
+                        external: {
+                            id: account.id,
+                            userName: nickname,
+                            userLogin: null,
```

Comment:
I think we can also safely assume:
```suggestion
                            userLogin: nickname,
```